core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/common/WorkflowActor.scala

```diff
@@ -2,19 +2,25 @@ package edu.uci.ics.amber.engine.architecture.common
 
 import akka.actor.{Actor, ActorLogging, Stash}
 import com.softwaremill.macwire.wire
-import edu.uci.ics.amber.engine.architecture.messaginglayer.MessagingManager
-import edu.uci.ics.amber.engine.architecture.receivesemantics.FIFOAccessPort
-import edu.uci.ics.amber.engine.common.ambermessage.WorkerMessage.DataMessage
+import edu.uci.ics.amber.engine.architecture.messaginglayer.{
+  ControlInputPort,
+  ControlOutputPort,
+  NetworkOutputGate
+}
+import edu.uci.ics.amber.engine.common.ambertag.neo.Identifier
 
-class WorkflowActor extends Actor with ActorLogging with Stash {
+class WorkflowActor(identifier: Identifier)
+    extends Actor
+    with ActorLogging
+    with Stash
+    with NetworkOutputGate {
 
-  lazy val input: FIFOAccessPort = wire[FIFOAccessPort]
-  lazy val messagingManager: MessagingManager = wire[MessagingManager]
+  val networkOutput: NetworkOutputGate = this
+  lazy val controlInputChannel: ControlInputPort = wire[ControlInputPort]
+  lazy val controlOutputChannel: ControlOutputPort = wire[ControlOutputPort]
```

Comment:
a way to abstract NetworkMessage from Processor is to put the dataInputChannel here and not in Processor.